src/Roslyn.Diagnostics.Analyzers/Core/ExposeMemberForTestingFixer.cs

```diff
@@ -0,0 +1,180 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Roslyn.Diagnostics.Analyzers
+{
+    public abstract class ExposeMemberForTestingFixer : CodeFixProvider
+    {
+        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(RoslynDiagnosticIds.ExposeMemberForTestingRuleId);
+
+        public sealed override FixAllProvider GetFixAllProvider()
+        {
+            // This is a refactoring for one-off test accessor creation. Batch fixing is disabled.
```

Comment:
We should write this as a CodeRefactoringProvider as well.

Reply:
➡️ Covered by https://github.com/dotnet/roslyn-analyzers/pull/2164#discussion_r264713698